### PR TITLE
Improves installation instruction and usage sample

### PR DESCRIPTION
**Pull request by mbeaudru**
Based on @avigoldman github comment on the  [Using React in Markdown source](https://github.com/gatsbyjs/gatsby/issues/312#issuecomment-387608217) discussion :+1:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,19 +14,89 @@ then
...
+      .then(result => {
+        if (result.errors) {
+          return reject(result.errors);
+        }
+
+        // Create markdown pages.
+        result.data.allFile.edges.forEach(
+          ({ node: { absolutePath, relativeDirectory, name } }) => {
+            createPage({
+              path: `/${relativeDirectory}/${name}`,
```
Comment (nhducit) [inline README.md:69]: I think it should be: "${relativeDirectory}/${name}" without the "/"

File: README.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## Installation
 
+### Dependencies
+
 ```
 yarn add gatsby-plugin-mdx
```
Comment (nhducit) [inline README.md:8]: Can you help to add `gatsby-source-filesystem` also?

**Comment by nhducit** [comment]
thanks @mbeaudru, I did exactly what you recommend in basic example: https://github.com/nhducit/gatsby-plugin-mdx/tree/master/examples/basic

File: README.md (hunk 1 above)
Comment (emilyhorsman) [reply README.md:69]: Trying this out in my own site, I agree.

**Comment by ChristopherBiscardi** [comment]
Work has moved to https://github.com/ChristopherBiscardi/gatsby-mdx for gatsby 2.0, which is close to final release.

**Pull request closed without merging**